### ISSUE TITLE
[REQUEST] Include and build mamba as a feedstock

### ISSUE THREAD

**Issue by alexlang74** [open-ce/open-ce#558]
`mamba` is faster than `conda`, so some teams want to switch to that. Unfortunately, the mamba packages on conda-forge have dependencies that clash with packages required by conda from the Anaconda main channel. 
The request is for the Open-CE team build the mamba-feedstock with dependencies from Anaconda main, across x86, ppc and s390.

**Comment by rolweber** [comment]
These are the packages that mamba 0.18.1 from conda-forge pulls in, for which no version exists on Anaconda main:
```txt
The following NEW packages will be INSTALLED:

  libmamba           conda-forge/linux-64::libmamba-0.18.1-h3985d26_1
  libmambapy         conda-forge/linux-64::libmambapy-0.18.1-py39h8bfa403_1
  libsolv            conda-forge/linux-64::libsolv-0.7.19-h780b84a_5
  libzlib            conda-forge/linux-64::libzlib-1.2.11-h36c2ea0_1013
  llvm-openmp        conda-forge/linux-64::llvm-openmp-12.0.1-h4bd325d_1
  mamba              conda-forge/linux-64::mamba-0.18.1-py39hfa8f2c8_1
  pybind11-abi       conda-forge/noarch::pybind11-abi-4-hd8ed1ab_3
  python_abi         conda-forge/linux-64::python_abi-3.9-2_cp39
  reproc             conda-forge/linux-64::reproc-14.2.3-h7f98852_0
  reproc-cpp         conda-forge/linux-64::reproc-cpp-14.2.3-h9c3ff4c_0
  yaml-cpp           conda-forge/linux-64::yaml-cpp-0.6.3-he1b5a44_4
```

llvm-openmp might be a dependency of the newer libstdcxx-ng which is pulled in from conda-forge. The others look to me as if they're directly related to the mamba functionality.

The packages I installed from Anaconda main...
<details><pre>
name: Mamba
channels:
  - defaults
dependencies:
  - _libgcc_mutex=0.1=main
  - _openmp_mutex=4.5=1_gnu
  - brotlipy=0.7.0=py39h27cfd23_1003
  - bzip2=1.0.8=h7b6447c_0
  - c-ares=1.17.1=h27cfd23_0
  - ca-certificates=2021.10.26=h06a4308_2
  - certifi=2021.10.8=py39h06a4308_0
  - cffi=1.14.6=py39h400218f_0
  - charset-normalizer=2.0.4=pyhd3eb1b0_0
  - conda=4.10.3=py39h06a4308_0
  - conda-package-handling=1.7.3=py39h27cfd23_1
  - cryptography=3.4.7=py39hd23ed53_0
  - icu=58.2=he6710b0_3
  - idna=3.2=pyhd3eb1b0_0
  - krb5=1.19.2=hac12032_0
  - ld_impl_linux-64=2.35.1=h7274673_9
  - libarchive=3.4.2=h62408e4_0
  - libcurl=7.78.0=h0b77cf5_0
  - libedit=3.1.20210910=h7f8727e_0
  - libev=4.33=h7f8727e_1
  - libffi=3.3=he6710b0_2
  - libgcc-ng=9.3.0=h5101ec6_17
  - libgomp=9.3.0=h5101ec6_17
  - libiconv=1.15=h63c8f33_5
  - libnghttp2=1.46.0=hce63b2e_0
  - libssh2=1.9.0=h1ba5d50_1
  - libstdcxx-ng=9.3.0=hd4cf53a_17
  - libxml2=2.9.12=h03d6c58_0
  - lz4-c=1.9.3=h295c915_1
  - lzo=2.10=h7b6447c_2
  - ncurses=6.3=h7f8727e_2
  - openssl=1.1.1l=h7f8727e_0
  - pip=21.2.4=py39h06a4308_0
  - pycosat=0.6.3=py39h27cfd23_0
  - pycparser=2.21=pyhd3eb1b0_0
  - pyopenssl=21.0.0=pyhd3eb1b0_1
  - pysocks=1.7.1=py39h06a4308_0
  - python=3.9.7=h12debd9_1
  - readline=8.1=h27cfd23_0
  - requests=2.26.0=pyhd3eb1b0_0
  - ruamel_yaml=0.15.100=py39h27cfd23_0
  - setuptools=58.0.4=py39h06a4308_0
  - six=1.16.0=pyhd3eb1b0_0
  - sqlite=3.36.0=hc218d9a_0
  - tk=8.6.11=h1ccaba5_0
  - tqdm=4.62.3=pyhd3eb1b0_1
  - tzdata=2021e=hda174b7_0
  - urllib3=1.26.7=pyhd3eb1b0_0
  - wheel=0.37.0=pyhd3eb1b0_1
  - xz=5.2.5=h7b6447c_0
  - yaml=0.2.5=h7b6447c_0
  - zlib=1.2.11=h7b6447c_3
  - zstd=1.4.9=haebb681_0
</details></pre>

Installation plan for adding mamba from conda-forge...
```bash
conda install --dry-run -c conda-forge mamba=0.18.1
```
<details><pre>
The following NEW packages will be INSTALLED:

  libmamba           conda-forge/linux-64::libmamba-0.18.1-h3985d26_1
  libmambapy         conda-forge/linux-64::libmambapy-0.18.1-py39h8bfa403_1
  libsolv            conda-forge/linux-64::libsolv-0.7.19-h780b84a_5
  libzlib            conda-forge/linux-64::libzlib-1.2.11-h36c2ea0_1013
  llvm-openmp        conda-forge/linux-64::llvm-openmp-12.0.1-h4bd325d_1
  mamba              conda-forge/linux-64::mamba-0.18.1-py39hfa8f2c8_1
  pybind11-abi       conda-forge/noarch::pybind11-abi-4-hd8ed1ab_3
  python_abi         conda-forge/linux-64::python_abi-3.9-2_cp39
  reproc             conda-forge/linux-64::reproc-14.2.3-h7f98852_0
  reproc-cpp         conda-forge/linux-64::reproc-cpp-14.2.3-h9c3ff4c_0
  yaml-cpp           conda-forge/linux-64::yaml-cpp-0.6.3-he1b5a44_4

The following packages will be REMOVED:

  libgomp-9.3.0-h5101ec6_17

The following packages will be UPDATED:

  certifi            pkgs/main::certifi-2021.10.8-py39h06a~ --> conda-forge::certifi-2021.10.8-py39hf3d152e_1
  libarchive         pkgs/main::libarchive-3.4.2-h62408e4_0 --> conda-forge::libarchive-3.5.2-hccf745f_1
  libcurl              pkgs/main::libcurl-7.78.0-h0b77cf5_0 --> conda-forge::libcurl-7.80.0-h2574ce0_0
  libgcc-ng          pkgs/main::libgcc-ng-9.3.0-h5101ec6_17 --> conda-forge::libgcc-ng-11.2.0-h1d223b6_11
  libiconv              pkgs/main::libiconv-1.15-h63c8f33_5 --> conda-forge::libiconv-1.16-h516909a_0
  libssh2               pkgs/main::libssh2-1.9.0-h1ba5d50_1 --> conda-forge::libssh2-1.10.0-ha56f1ee_2
  libstdcxx-ng       pkgs/main::libstdcxx-ng-9.3.0-hd4cf53~ --> conda-forge::libstdcxx-ng-11.2.0-he4da1e4_11
  zlib                    pkgs/main::zlib-1.2.11-h7b6447c_3 --> conda-forge::zlib-1.2.11-h36c2ea0_1013
  zstd                     pkgs/main::zstd-1.4.9-haebb681_0 --> conda-forge::zstd-1.5.0-ha95c52a_0

The following packages will be SUPERSEDED by a higher-priority channel:

  _libgcc_mutex           pkgs/main::_libgcc_mutex-0.1-main --> conda-forge::_libgcc_mutex-0.1-conda_forge
  _openmp_mutex          pkgs/main::_openmp_mutex-4.5-1_gnu --> conda-forge::_openmp_mutex-4.5-1_llvm
  ca-certificates    pkgs/main::ca-certificates-2021.10.26~ --> conda-forge::ca-certificates-2021.10.8-ha878542_0
  openssl              pkgs/main::openssl-1.1.1l-h7f8727e_0 --> conda-forge::openssl-1.1.1l-h7f98852_0
</details></pre>

**Comment by cdeepali** [comment]
Mamba is available in OpenCE v1.5.2 onwards.
